js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx

```diff
@@ -0,0 +1,67 @@
+import {DocumentNode, OperationVariables, useApolloClient} from '@apollo/client';
+import {cache} from 'idb-lru-cache';
+import React from 'react';
+
+const ONE_WEEK = 1000 * 60 * 60 * 24 * 7;
+
+/**
+ * Returns data from the indexedDB cache initially while loading is true.
+ * Fetches data from the network/cache initially and does not receive any updates afterwards
+ */
+export function useIndexedDBCachedQuery<TQuery, TVariables extends OperationVariables>({
+  key,
+  query,
+  variables,
+}: {
+  key: string;
+  query: DocumentNode;
+  variables: TVariables;
```

Comment:
Make this optional? Looks like right now we'll have a new `{}` on every call to this hook, which would affect the deps list on the `fetch` callback, even though neither query has variables. (And it would be kind of annoying to have to provide a memoized `{}` for each hook callsite.)

It's also possible that `initialize` in `useGlobalSearch` doesn't really need to be a `callback`, and that the fetch functions don't need to be either.

Reply:
The variables don't actually end up mattering because it's only used in the useCallback